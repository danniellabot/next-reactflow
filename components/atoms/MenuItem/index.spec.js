// test MenuItemComponent renders correctly
// test onClick event is handled using fireEvent
// props object

import React from 'react';
import renderer from 'react-test-renderer';
import { render, fireEvent } from '@testing-library/react';
import MenuItemComponent from './index';

describe('MenuItemComponent', () => {
    it('renders correctly', () => {
        const tree = renderer.create(<MenuItemComponent />).toJSON();
        expect(tree).toMatchSnapshot();
    });

    it('handles onClick event', () => {
        const props = {
            label: 'Test',
            onClick: jest.fn(),
        };
        const { getByText } = render(<MenuItemComponent {...props} />);
        fireEvent.click(getByText(props.label));
        expect(props.onClick).toHaveBeenCalled();
    });
}
)




