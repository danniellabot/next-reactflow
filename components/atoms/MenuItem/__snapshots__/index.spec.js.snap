// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MenuItemComponent renders correctly 1`] = `
<li
  className="MuiButtonBase-root MuiMenuItem-root MuiMenuItem-gutters css-kk1bwy-MuiButtonBase-root-MuiMenuItem-root"
  onBlur={[Function]}
  onContextMenu={[Function]}
  onDragLeave={[Function]}
  onFocus={[Function]}
  onKeyDown={[Function]}
  onKeyUp={[Function]}
  onMouseDown={[Function]}
  onMouseLeave={[Function]}
  onMouseUp={[Function]}
  onTouchEnd={[Function]}
  onTouchMove={[Function]}
  onTouchStart={[Function]}
  role="menuitem"
  tabIndex={-1}
/>
`;
